google/resource_google_project_iam_policy.go

```diff
@@ -31,8 +31,9 @@ func resourceGoogleProjectIamPolicy() *schema.Resource {
 				DiffSuppressFunc: jsonPolicyDiffSuppress,
 			},
 			"authoritative": &schema.Schema{
-				Type:     schema.TypeBool,
-				Optional: true,
+				Type:       schema.TypeBool,
+				Optional:   true,
+				Deprecated: "Use google_project_iam_policy_binding and google_project_iam_policy_member instead.",
```

Comment:
Quick sanity check on this for me if I may. Right now `google_project_iam_policy` defaults attempting to merge policies, and the intent is in the future to make it authoritative for policy management on a project (ie, remove this code, and have the code always call setIamPolicy instead of attempt merging). Today, if I wanted that behaviour, I'd set `authoritative = true`, right? Right now when I do that, I'm faced with this deprecation warning here, and I just want to make sure that's intentional.

Reply:
That's correct. When 2.0.0 ships, the resource will stick around, and there will be no "authoritative" field; it will just always control the entire IAM policy for the project.

Unfortunately, we have no way to silence the deprecation warning if it's set to "true", and I'm not sure we'd want to if we could, because you'll still need to delete the line from your config when you upgrade to 2.0.0. But I'll see if I can make the deprecation message a bit clearer.

Reply:
I figured that was the case - and it makes sense to me.. Just wanted to make sure I wasn't backing myself into a corner. Thanks! 🙏

Reply:
I opened #1343 to make this a bit clearer.